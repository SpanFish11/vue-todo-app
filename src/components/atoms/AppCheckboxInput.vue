<script setup>
const props = defineProps({
  disabled: { type: Boolean, required: false, default: false },
  checked: { type: Boolean, required: false, default: false },
})

const emit = defineEmits(['onChange'])
</script>
`
<template>
  <input
    class="form-check-input"
    type="checkbox"
    :disabled="props.disabled"
    :checked="props.checked"
    @change="emit('onChange')"
  />
</template>

<style scoped lang="scss">
.form-check-input {
  border: transparent;
  width: 23px;
  height: 23px;
  background: #c0c0c0;
  -webkit-mask: url('../../assets/icons/circle.svg') no-repeat center;
  -webkit-mask-size: contain;
  mask: url('../../assets/icons/circle.svg') no-repeat center;
  mask-size: contain;
  stroke: none;

  &:focus {
    box-shadow: none;
  }

  &:checked {
    border: none;
    width: 23px;
    height: 23px;
    background: linear-gradient(130deg, rgba(25, 134, 217, 0.65), rgba(200, 57, 222, 0.72));
    -webkit-mask: url('../../assets/icons/check-circle-fill.svg') no-repeat center;
    -webkit-mask-size: contain;
    mask: url('../../assets/icons/check-circle-fill.svg') no-repeat center;
    mask-size: contain;
  }

  &:hover {
    cursor: pointer;
    background: linear-gradient(130deg, rgba(38, 121, 180, 0.6526260162268032), rgba(189, 58, 208, 0.7198529069831057));
  }
}
</style>
