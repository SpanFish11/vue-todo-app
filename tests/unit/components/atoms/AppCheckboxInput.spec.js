import { beforeEach, describe, expect, it } from '@jest/globals'
import { shallowMount } from '@vue/test-utils'
import AppCheckboxInput from '@/components/atoms/AppCheckboxInput.vue'

describe('Given the AppCheckboxInput component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(AppCheckboxInput, {})
  })

  it.each([
    [{ disabled: false, checked: false }],
    [{ disabled: true, checked: false }],
    [{ disabled: false, checked: true }],
    [{}],
  ])('should render', (props) => {
    wrapper.setProps(props)

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should emits a click event when input is clicked', async () => {
    await wrapper.find('input').trigger('change')
    const onClickEvent = wrapper.emitted('onChange')

    expect(wrapper.emitted()).toHaveProperty('onChange')
    expect(onClickEvent).toHaveLength(1)
  })
})
