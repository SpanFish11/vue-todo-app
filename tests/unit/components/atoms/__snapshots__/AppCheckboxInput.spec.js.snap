// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AppCheckboxInput component should render 1`] = `
<input
  class="form-check-input"
  type="checkbox"
/>
`;

exports[`Given the AppCheckboxInput component should render 2`] = `
<input
  class="form-check-input"
  type="checkbox"
/>
`;

exports[`Given the AppCheckboxInput component should render 3`] = `
<input
  class="form-check-input"
  type="checkbox"
/>
`;

exports[`Given the AppCheckboxInput component should render 4`] = `
<input
  class="form-check-input"
  type="checkbox"
/>
`;
